ThreeUpsThenDownBot.cs

```diff
@@ -68,24 +73,42 @@ public ThreeUpsThenDownBot(Decimal initialFunds)
       // can't sell if I have no coins
       if (CoinCount <= 0m) return null;
       
-      // if at any time the price drops below 5% of where I bought, then sell as a safety measure
+      // I don't know why these would be null... I just gotta ignore these scenarios for now...
       if (CurrentPrice == null) return null;
-      if (CurrentPrice.Value * 0.95m > LastPurchasedPerCoinPrice.Value) return CoinCount;
+      if (LastPurchasedPerCoinPrice == null) return null;
       
-      // if at any time the price drops below 50% of the difference 
-      // from "where I bought it" to "the max I've possibly been able to buy it"
-      // then sell as a profit-guaranteeing measure
-      if (MaxObservedPotentialSalePrice != null &&
-          (MaxObservedPotentialSalePrice.Value - LastPurchasedPerCoinPrice.Value) * 0.5m + LastPurchasedPerCoinPrice.Value > CurrentPrice.Value)
+      // if at any time the price drops below 5% of where I bought, then sell as a safety measure
+      if (CurrentPrice.Value * 0.95m > LastPurchasedPerCoinPrice.Value)
       {
+        _logger.LogInformation($"Selling because CurrentPrice ({CurrentPrice}) * 0.95m > LastPurchasedPerCoinPrice ({LastPurchasedPerCoinPrice})");
```

Comment:
The logger is a decent "debugging" way to reveal this info, but I would prefer a more formal way for bots to report decisions they're making that doesn't fill the console output with waste